### ISSUE TITLE
Document default iput/put API behaviour with large files

### ISSUE THREAD

**Issue by kript** [irods/irods#4561]
### iRODS Version, OS and Version

iRODs 4.1.12

### What did you try to do?

Upload a file (via `iput` or (baton)[https://github.com/wtsi-npg/baton] larger than 32MB from a client that can access an IES but is restricted via firewall ACL's to get to the iRES's.

`-I` was *not* used

### Expected behavior

File is uploaded, going via IES rather than direct to the resource owning iRES's

### Observed behavior (including steps to reproduce, if applicable)

Timout for upload with 
```
ERROR: connectToRhostPortal: connectTo Rhost
irods-resource-server.network.sanger.ac.uk port 20130 error, status = -347000
ERROR: putUtil: put error for /seq/npg/test/test.txt, status = -347000 status =
-347000 USER_SOCK_CONNECT_TIMEDOU
```
IES log is;
```
Oct 14 11:26:44 pid:326 NOTICE: connectToRhost: connect to host irods-resource-server.network.sanger.ac.uk on port 1247 failed status = -347000 USER_SOCK_CONNECT_TIME
DOUT
Oct 14 11:26:44 pid:326 ERROR: _rcConnect: connectToRhost error, server onirods-resource-server.network.sanger.ac.uk:1247 is probably down status = -347000 USER_SOCK
_CONNECT_TIMEDOUT
Oct 14 11:26:44 pid:326 ERROR: _rsZoneReport - get_server_reports failed, status
 = -347000
Oct 14 11:26:44 pid:326 ERROR: rsZoneReport: rcZoneReport failed, status = -3470
00
Oct 14 11:26:44 pid:326 ERROR: [-]      iRODS/server/core/src/rsApiHandler.cpp:5
25:readAndProcClientMsg :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message
 []
        [-]     iRODS/lib/core/src/sockComm.cpp:199:readMsgHeader :  status [SYS
_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
                [-]     libtcp.cpp:197:tcp_read_msg_header :  status [SYS_HEADER
_READ_LEN_ERR]  errno [] -- message [only read [0] of [4]]
```

I expect that this is because the behaviour described by `-I` is in fact the default for files larger than 32MB for the put API, however, this not documented in the `iput` man page, and in fact implied otherwise by `-I` being documented but nothing else.

Relatedly, is there a way to _stop_ iput from going direct on 'large' file/parallel mode uploads?

**Comment by trel** [comment]
Yes, for large - No for parallel.   I believe you can use `-N0` and/or `-N1` to indicate no parallel threading, and that will honor the multi-hop path...

Otherwise, parallel always goes direct point-to-point.   We will update the `iput` documentation.

Soon, all data movement (big and small) will use a single code path and be point-to-point **AND ONLY ON PORT 1247** (https://github.com/irods/irods/issues/4336).   With that said, if a server with the new code is talking to a client or other server without the new code, the current/existing behavior will still be seen.

Are you requesting a mode whereby the client can request **NOT** to be point-to-point?  I think we can do that with the new code...

**Comment by trel** [comment]
Got it.

In the meantime, I think you can navigate the firewall by dropping to a single thread?

`iput -N1`... (which I think is equivalent to `iput -N0`)
